### ISSUE TITLE
Trace perform weirdly on BatchNorm

### ISSUE THREAD

**Issue by yaojunr** [pytorch/pytorch#18312]
## 🐛 Bug

I have use jit to trace a model which is based on ResNet. When the batch size of the input is 1, the result is correct. However, if the batch size is not 1, the result have a little difference. And I think it's caused by the different behavior of BatchNorm.

## To Reproduce
I use the code below.
```python
import torch
import torchvision

resnet = torchvision.models.resnet18(True)
resnet.eval()
traced_net = torch.jit.trace(resnet,torch.randn((3,3,224,224)))

traced_net.eval()

def check(inp):
    out1 = resnet(inp)
    out2 = traced_net(inp)
    return torch.sum(out1==out2)

for i in range(5):
    print(check(torch.randn((1,3,224,224))))
for i in range(5):
    print(check(torch.randn((3,3,224,224))))

#output
#tensor(1000)
#tensor(1000)
#tensor(1000)
#tensor(1000)
#tensor(1000)
#tensor(2009)
#tensor(2066)
#tensor(2119)
#tensor(2049)
#tensor(2023)
```

## Expected behavior

With the input shape of (3,3,224,224), the result should be 3*1000=3000, but it's just around 2000.

## Environment

PyTorch version: 1.0.1.post2
Is debug build: No
CUDA used to build PyTorch: 9.0.176

OS: Ubuntu 16.04.5 LTS
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.10) 5.4.0 20160609
CMake version: version 3.13.1

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 9.0.176

[pip3] numpy==1.14.3
[pip3] numpydoc==0.8.0
[pip3] torch==1.0.1.post2
[pip3] torchvision==0.2.0

**Comment by ailzhang** [comment]
It's more likely a precision issue. I will trace it down.

**Comment by tjsongzw** [comment]
`torch.testing.assert_allclose` works

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
